app/models/service_plan.rb

```diff
@@ -4,4 +4,10 @@ class ServicePlan < ApplicationRecord
 
   belongs_to :portfolio_item
   validates :base, :presence => true
+
+  validate :valid_ddf, :on => :update
```

Comment:
Possibly pull `valid_` out of the this name too? Possibly just call it `data_driven_form`? 
```
validate :data_driven_form, :on => :update
```
Is just as informational without the `valid_` in the ddf method name.

Reply:
Yeah I like that, changed all of them.